src/panels/lovelace/entity-rows/hui-input-number-entity-row.ts

```diff
@@ -0,0 +1,136 @@
+import { setValue } from "../../../data/input_text";
+
+@customElement("hui-input-number-entity-row")
+class HuiInputNumberEntityRow extends LitElement implements EntityRow {
+  @property() public hass?: HomeAssistant;
+  @property() private _config?: EntityConfig;
+
+  public setConfig(config: EntityConfig): void {
+    if (!config) {
+      throw new Error("Configuration error");
+    }
+    this._config = config;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row
+        .hass="${this.hass}"
+        .config="${this._config}"
+        id="input_number_card"
+      >
+        <div>
+          ${stateObj.attributes.mode === "slider"
+            ? html`
+                <div class="flex">
+                  <ha-slider
+                    .dir="${this._computeRTLDirection}"
+                    .step="${Number(stateObj.attributes.step)}"
+                    .min="${Number(stateObj.attributes.min)}"
+                    .max="${Number(stateObj.attributes.max)}"
+                    .value="${Number(stateObj.state)}"
+                    pin
+                    @change="${this._selectedValueChanged}"
+                    ignore-bar-touch
+                    id="input"
+                  ></ha-slider>
+                  <span class="state">
+                    ${Number(stateObj.state)}
+                    ${stateObj.attributes.unit_of_measurement}
+                  </span>
+                </div>
+              `
+            : html`
+                <paper-input
+                  no-label-float
+                  auto-validate
+                  .pattern="[0-9]+([\\.][0-9]+)?"
+                  .step="${Number(stateObj.attributes.step)}"
+                  .min="${Number(stateObj.attributes.min)}"
+                  .max="${Number(stateObj.attributes.max)}"
+                  .value="${Number(stateObj.state)}"
+                  type="number"
+                  @change="${this._selectedValueChanged}"
+                  id="input"
+                ></paper-input>
+              `}
+        </div>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      .flex {
+        display: flex;
+        align-items: center;
+      }
+      .state {
+        min-width: 45px;
+        text-align: center;
+      }
+      paper-input {
+        text-align: right;
+      }
+    `;
+  }
+
+  private get _inputElement(): PaperInputElement {
+    return this.shadowRoot!.getElementById("input") as PaperInputElement;
```

Comment:
This is not always true, as both have an ID set. I would instead just use `querySelector('paper-input')`, since we have just one.

Reply:
but only one will ever be rendered, the slider or the box

Reply:
Oh I see what you did there. Both have the same API so it works. I would suggest you just set the type to `{ value: string }`

Reply:
I don't follow

Reply:
Well, you say your return value is a PaperInputElement, but it might be that you are rendering a slider, so that's the wrong type. It only works because the parts of PaperInputElement that you access have the same API as PaperSliderElement. The overlapping part that you care about is `{ value: string }`, so I suggest that you replace the return value and type casting (`as`) to `{ value: string }`

Reply:
![image](https://user-images.githubusercontent.com/1287159/53055680-31ea5b00-346e-11e9-929d-a1f2fb831d92.png)

This is what linter wants me to do:
![image](https://user-images.githubusercontent.com/1287159/53055693-40d10d80-346e-11e9-82ea-fe4bcdbb4f0d.png)

Reply:
yeah that's fine.  just add a comment